python/lbann/models/transformer.py

```diff
@@ -393,6 +396,7 @@ def __init__(
             )
             for i in range(num_decoder_layers)
         ]
+        self.separate_heads = self.decoder[0].attention1.separate_heads
```

Comment:
Is it worth sanity-checking that all layers have the same `separate_heads` setting?

Reply:
Done